src/TorchSharp/NN/Module.cs

```diff
@@ -757,28 +756,51 @@ public virtual void register_parameter(string name, Parameter param)
                 }
 
                 /// <summary>
-                /// Register a submodule.
+                /// Adds a child module to the current module.
+                /// The module can be accessed as an attribute using the given name.
                 /// </summary>
-                /// <param name="name">Name of the submodule.</param>
-                /// <param name="submodule">The module to register.</param>
+                /// <param name="name">
+                /// name of the child module.
+                /// The child module can be accessed from this module using the given name
+                /// </param>
+                /// <param name="module">child module to be added to the module.</param>
+                /// <exception cref="ArgumentException"></exception>
                 /// <exception cref="InvalidOperationException"></exception>
-                public virtual void register_module(string name, Module submodule)
```

Comment:
Changing the name of a parameter of a public method is a breaking change, so unless it's absolutely necessary, it should be avoided.